docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
 However, there may be many problems when VPN is used to directly connect the components/services of different security levels:
 
 * It is too risky to directly expose the central cloud management network/IP to users
-
 * IP needs to be planned globally in advance to ensure that the management components/service IP of the central cloud and each edge site do not conflict
-
 * Network changes can be difficult
 
 This proposal aims to outline a design for cloud-edge communication framework/service to solve the above problems
 
 ## Goals
 
 * Support communication access capability of cloud-edge basic management components
-
 * Support independent IP addressing and IP address convergence of edge management components/services
-
 * Minimize attack surface, zero trust security
 
+## Proposal
+
+The following two solutions do not conflict with each other. The former can converge IP addresses to resolve IP address conflicts. However, the communication needs to be first sent to the gateway and then to the destination service. The latter cannot converge IP addresses but can resolve IP address conflicts. The latter provides end-to-end communication.
+
+### Proposal A
+
+The communication of cloud edge components/services is realized based on L4/L7 proxy, which is similar to kubedge's cloud-edge message channel.
+
+Cloudgateway and edgegateway are introduced into the central cloud and each edge site respectively. The edge site edgegateway supports connecting to the cloud gateway of the central cloud when it is started. The service communication between cloud and edge needs to go through these two gateways before arriving. Service access interfaces are provided on these two gateways to receive requests and route requests to target services.
+
+* Cloudgateway and edgegateway are installed and run in the central cloud and each edge site respectively
+* Edgegateway of Edge site can actively establish encrypted links with cloudgateway on the central side, and support two-way message transmission
+* Register the services of edge site and central cloud to cloudgateway
+* Edgegateway in the edge site acts as the service in the central cloud, and exposes the service access interface of the central cloud to the service in the edge site
```

Comment:
I'm trying to understand the difference between option A and B.  In option A it's also mentioned "...exposes the service access interface of the central cloud to the service in the edge site". What's the difference of "service interface" here than the virtual presentation in option B? Is the service interface in A only network-level?

Reply:
Service access interface just mean the api endpoint in option A must be corrected.

In simple terms, Option A works at a higher network layer (Layer 4/Layer 7). It is similar to Nginx. It provides backend service agents and transfers cloud or edge sites to other sites.

Option B is a type of (no man-in-the-middle, no session termination) and Virtual Presence is a local virtual presentation of remote services, It can provide full isolation of remote servers, it use of locally assigned virtual presence IPs and External IPs to avoid exposure of site internal addresses. that work at the Layer 3 network layer.
we want to use translation mechanisms to emulate local traffic flow, to implement it.

Reply:
"Edgegateway in the edge site acts as the service in the central cloud, and exposes the service access interface of the central cloud to the service in the edge site" this is quite hard to understand.

what's central cloud? why the word "central"? is it just the cloud?
"acts as the service" --> "acts as a service"
if "service access interface" means api, can we just say API?
if edgegateway acts as a service in the cloud, how can it expose API of cloud to service in the edge site?

consider add some more explanation or an example for this. I still can't understand it.

Reply:
what's central cloud? why the word "central"? is it just the cloud?
-- yes, it is just the cloud

"acts as the service" --> "acts as a service"
if "service access interface" means api, can we just say API?
if edgegateway acts as a service in the cloud, how can it expose API of cloud to service in the edge site?
-- In proposal A, edgegateway acts as the proxy of the services running in the cloud

consider add some more explanation or an example for this. I still can't understand it.
-- ok, I will add some example for it